superset/db_engine_specs/impala.py

```diff
@@ -43,11 +43,13 @@ def epoch_to_dttm(cls) -> str:
         return "from_unixtime({col})"
 
     @classmethod
-    def convert_dttm(cls, target_type: str, dttm: datetime) -> str:
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
         tt = target_type.upper()
         if tt == "DATE":
-            return "'{}'".format(dttm.strftime("%Y-%m-%d"))
-        return "'{}'".format(dttm.strftime("%Y-%m-%d %H:%M:%S"))
+            return f"CAST('{dttm.date().isoformat()}' AS DATE)"
+        elif tt == "TIMESTAMP":
+            return f"""CAST('{dttm.isoformat(sep=" ", timespec="seconds")}' AS TIMESTAMP)"""  # pylint: disable=line-too-long
```

Comment:
It seems Impala is ok with the "T" separator, and should support nanosecond precision, i.e. should be ok with `dttm.isoformat()`: https://impala.apache.org/docs/build/html/topics/impala_timestamp.html